### PR TITLE
Add new tests to create new contact

### PR DESCRIPTION
**Pull request by venureddy95**
Added 6 new scenarios including a page object and step definitions.

### CODE REVIEW

File: test/acceptance/features/create-a-contact.feature [hunk 1]
```diff
@@ -0,0 +1,30 @@
+@contact-create
+Feature: Create New Contact
+  As a existing user
```
Comment (tyom) [inline test/acceptance/features/create-a-contact.feature]: an existing user

File: test/acceptance/page_objects/Contact.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
+// const shortId = require('shortid')
+// const { getRandomNumber } = require('../../helper')
```
Comment (tyom) [inline test/acceptance/page_objects/Contact.js]: I'd suggest removing the commented out lines.

File: test/acceptance/page_objects/Contact.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    findUkAddressButton: {
+      selector: ".//*[@id='address-wrapper']/div[2]/button",
+      locateStrategy: 'xpath',
+    },
+    selectUkAddressDropdown: {
+      selector: ".//*[@id='address-wrapper']/div[3]/select",
+      locateStrategy: 'xpath',
+    },
+    selectAnUkAddressFromList: '#address-wrapper div:nth-child(3) select option:nth-child(3)',
+    // selectAnUkAddressFromList: {
```
Comment (tyom) [inline test/acceptance/page_objects/Contact.js]: Can remove this as the line above does the same.

File: test/acceptance/page_objects/Contact.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        return this
+          .setValue('#search', name)
+          .submitForm('@searchForm')
+      },
+
+      navigateToContactsPage () {
+        return this
+          .click('@firstCompanyFromList')
+          .click('@contactsTab')
+          // .api.pause()
```
Comment (tyom) [inline test/acceptance/page_objects/Contact.js]: Remove.

File: test/acceptance/page_objects/Contact.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+          .setValue('@contactTelephoneCountryCode', faker.random.number())
+          .setValue('@contactTelephoneNumber', faker.phone.phoneNumberFormat())
+          .setValue('@contactEmailAddress', firstName + lastName + '@dit.gov.uk')
+          .click('@contactSameAddressNo')
+          .setValue('@ukPostcode', 'EC2Y 9AE')
+          .click('@findUkAddressButton')
+          .click('@selectUkAddressDropdown')
+          .waitForElementPresent('@selectAnUkAddressFromList', 5000)
+          .click('@selectAnUkAddressFromList')
+          // .api.keys(this.api.keys.TAB)
```
Comment (tyom) [inline test/acceptance/page_objects/Contact.js]: Remove.

File: test/acceptance/page_objects/Contact.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+          .setValue('@contactLastname', lastName)
+          .setValue('@contactJobTitle', faker.name.jobTitle())
+          .click('@contactPrimaryContactYes')
+          .setValue('@contactTelephoneCountryCode', faker.random.number())
+          .setValue('@contactTelephoneNumber', faker.phone.phoneNumberFormat())
+          .setValue('@contactEmailAddress', firstName + lastName + '@dit.gov.uk')
+          .setValue('@contactAlternatePhonenumber', '666555444')
+          .setValue('@contactAlternativeEmail', lastName + firstName + '@dit.gov.uk')
+          .setValue('@contactNotes', faker.name.jobDescriptor() + firstName)
+          .api.pause()
```
Comment (tyom) [inline test/acceptance/page_objects/Contact.js]: Remove pause.

File: test/acceptance/features/create-a-contact.feature (hunk 1 above)
Comment (venureddy95) [inline test/acceptance/features/create-a-contact.feature]: Changes implemented

File: test/acceptance/page_objects/Contact.js (hunk 2 above)
Comment (venureddy95) [inline test/acceptance/page_objects/Contact.js]: Removed.

File: test/acceptance/page_objects/Contact.js (hunk 3 above)
Comment (venureddy95) [inline test/acceptance/page_objects/Contact.js]: Removed.

File: test/acceptance/page_objects/Contact.js (hunk 4 above)
Comment (venureddy95) [inline test/acceptance/page_objects/Contact.js]: Removed

File: test/acceptance/page_objects/Contact.js (hunk 6 above)
Comment (venureddy95) [inline test/acceptance/page_objects/Contact.js]: aha... this was the one which was stopping the test and making it timeout. Removed now.

File: test/acceptance/page_objects/Contact.js (hunk 5 above)
Comment (venureddy95) [inline test/acceptance/page_objects/Contact.js]: Removed.

**Comment by feedmypixel** [comment]
@venureddy95 its worth noting we use `git commit --fixup <sha>` when adding work from PR comments. It makes it much easier to see what has changed. Have a read up in the git docs about it. And don't forget to squash before merge (said from experience).

File: test/acceptance/features/step_definitions/create-a-contact.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
+const shortId = require('shortid')
+const { client } = require('nightwatch-cucumber')
+const { defineSupportCode } = require('cucumber')
+
+defineSupportCode(({ Given, Then, When }) => {
+  const Company = client.page.Company()
+  const Contact = client.page.Contact()
+  const randomCompanyName = `TestingNewCompany-1206-1`
```
Comment (feedmypixel) [inline test/acceptance/features/step_definitions/create-a-contact.js]: doesn't need to be a template literal

File: test/acceptance/page_objects/Contact.js [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+      createNewPrimaryContact (firstName, lastName) {
+        return this
+          .click('@addContactButton')
+          .setValue('@contactFirstname', firstName)
+          .setValue('@contactLastname', lastName)
+          .setValue('@contactJobTitle', faker.name.jobTitle())
+          .click('@contactPrimaryContactNo')
+          .setValue('@contactTelephoneCountryCode', faker.random.number())
+          .setValue('@contactTelephoneNumber', faker.phone.phoneNumberFormat())
+          .setValue('@contactEmailAddress', firstName + lastName + '@dit.gov.uk')
```
Comment (feedmypixel) [inline test/acceptance/page_objects/Contact.js]: these could be template literals

File: test/acceptance/page_objects/Contact.js [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+          .setValue('@contactAlternatePhonenumber', '666555444')
+          .setValue('@contactAlternativeEmail', lastName + firstName + '@dit.gov.uk')
```
Comment (feedmypixel) [inline test/acceptance/page_objects/Contact.js]: these could be template literals

File: test/acceptance/page_objects/Contact.js [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+      createNewNonPrimaryContact (firstName, lastName) {
+        return this
+          .click('@addContactButton')
+          .setValue('@contactFirstname', firstName)
+          .setValue('@contactLastname', lastName)
+          .setValue('@contactJobTitle', faker.name.jobTitle())
+          .click('@contactPrimaryContactYes')
+          .setValue('@contactTelephoneCountryCode', faker.random.number())
+          .setValue('@contactTelephoneNumber', faker.phone.phoneNumberFormat())
+          .setValue('@contactEmailAddress', firstName + lastName + '@dit.gov.uk')
```
Comment (feedmypixel) [inline test/acceptance/page_objects/Contact.js]: these could be template literals

File: test/acceptance/page_objects/Contact.js [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+          .setValue('@contactAlternatePhonenumber', '666555444')
+          .setValue('@contactAlternativeEmail', lastName + firstName + '@dit.gov.uk')
```
Comment (feedmypixel) [inline test/acceptance/page_objects/Contact.js]: these could be template literals

File: test/acceptance/page_objects/Contact.js [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+      createNewPrimaryContactWithNewCompanyAddress (firstName, lastName) {
+        return this
+          .click('@addContactButton')
+          .setValue('@contactFirstname', firstName)
+          .setValue('@contactLastname', lastName)
+          .setValue('@contactJobTitle', faker.name.jobTitle())
+          .click('@contactPrimaryContactNo')
+          .setValue('@contactTelephoneCountryCode', faker.random.number())
+          .setValue('@contactTelephoneNumber', faker.phone.phoneNumberFormat())
+          .setValue('@contactEmailAddress', firstName + lastName + '@dit.gov.uk')
```
Comment (feedmypixel) [inline test/acceptance/page_objects/Contact.js]: these could be template literals

File: test/acceptance/page_objects/Contact.js [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+          .click('@contactSameAddressNo')
+          .setValue('@ukPostcode', 'EC2Y 9AE')
+          .click('@findUkAddressButton')
+          .click('@selectUkAddressDropdown')
+          .waitForElementPresent('@selectAnUkAddressFromList', 5000)
+          .click('@selectAnUkAddressFromList')
+          // .api.keys(this.api.keys.TAB)
+          .setValue('@contactAlternatePhonenumber', '666555444')
+          .setValue('@contactAlternativeEmail', lastName + firstName + '@dit.gov.uk')
```
Comment (feedmypixel) [inline test/acceptance/page_objects/Contact.js]: these could be template literals

File: test/acceptance/features/step_definitions/create-a-contact.js (hunk 7 above)
Comment (tyom) [inline test/acceptance/features/step_definitions/create-a-contact.js]: Also, should the id be generated?

File: test/acceptance/page_objects/Contact.js (hunk 8 above)
Comment (feedmypixel) [inline test/acceptance/page_objects/Contact.js]: For example 
`` `${firstName + lastName}@dit.gov.uk` ``
or 
`` `${firstName}${lastName}@dit.gov.uk` ``

**Comment by feedmypixel** [comment]
> @venureddy95 its worth noting we use git commit --fixup <sha> when adding work from PR comments. It makes it much easier to see what has changed. Have a read up in the git docs about it. And don't forget to squash before merge (said from experience).

@venureddy95 If you would like some pointers on how to do this give me a shout

File: test/acceptance/features/step_definitions/create-a-interaction-step-definitions.js [hunk 14]
```diff
@@ -0,0 +1,240 @@
...
+    await Company
+      .navigate()
+      .findCompany(randomCompanyName)
+    await Contact
+      .clickOnFirstCompanyFromList()
+    await Interaction
+      .navigateToInteractionsPage()
+      .clickAddInteractionButton()
+      .clickServiceDeliveryRadioButton()
+      // .enterNewInteractionDetails(subject)
```
Comment (tyom) [inline test/acceptance/features/step_definitions/create-a-interaction-step-definitions.js]: Remove?

**Comment by feedmypixel** [comment]
A few little things but it feels like these would be best to catch up on in person. With all the current refactoring its much more valuable to have more acceptance tests in the code base

File: test/acceptance/page_objects/Contact.js [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+    selectUkAddressDropdown: {
+      selector: ".//*[@id='address-wrapper']/div[3]/select",
+      locateStrategy: 'xpath',
+    },
+    selectAnUkAddressFromList: '#address-wrapper div:nth-child(3) select option:nth-child(3)',
+  },
+
+  commands: [
+    {
+      findCompany (name) {
```
Comment (tyom) [inline test/acceptance/page_objects/Contact.js]: Feels like this should be under `Company` page object.

File: test/acceptance/page_objects/Contact.js [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+        return this
+          .setValue('#search', name)
+          .submitForm('@searchForm')
+      },
+
+      clickOnFirstCompanyFromList () {
```
Comment (tyom) [inline test/acceptance/page_objects/Contact.js]: This as well might be concern of the Company page object.

**Pull request merged**